packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -134,7 +145,13 @@ const isMultiSig = (tx: Transaction): boolean => {
 type LedgerConnectionWithCommunicationTypeAndDevicePath = {
   deviceConnection: LedgerConnection;
   communicationType: CommunicationType;
-  devicePath?: string;
+  nodeHidDevicePath?: string;
+  device?: LedgerDevice;
```

Comment:
Added a device object to the remembered connection data in order to retrieve existing connection for the particular device. Recognition is done by a shallow comparison of the device objects.

Reply:
Same as https://github.com/input-output-hk/cardano-js-sdk/pull/1161/files#r1540811427